### PR TITLE
Add references to state

### PR DESCRIPTION
**Pull request by smdawe**

### CODE REVIEW

**Comment by smdawe** [comment]
@domnikl I have this PR ready.

File: examples/schema-registry.yml [hunk 1]
```diff
@@ -15,3 +15,11 @@ subjects:
   - name: json-schema-value
     file: foo.json
     type: JSON
+
+  - name: with-references
+    file: with-references.avsc
+    type: AVRO
+    references:
+      - name: dev.domnikl.schema_registry_gitop.User
+        subject: User
```
Comment (domnikl) [inline examples/schema-registry.yml:24]: Subject is named `User-value` above.

File: examples/with-references.avsc [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+      "type": "record",
+      "name": "UserReference",
+      "namespace": "dev.domnikl.schema_registry_gitops",
+      "fields": [
+            {
+                  "name": "user",
+                  "type": "dev.domnikl.schema_registry_gitops.User"
+            }
+      ]
+}
```
Comment (domnikl) [inline examples/with-references.avsc:11]: Please add a newline at end of file.

**Review by domnikl — CHANGES_REQUESTED** [review]
Big thank you for the PR @smdawe! A commented a few little things that need changes. Also this should be documented in the README, but I can also do that afterwards.

**Comment by smdawe** [comment]
@domnikl thanks for your feedback.  All issues raised have been addressed.  This has been my first dip into kotlin and I have really enjoyed what I have seen so far.

**Review by domnikl — APPROVED** [review]
@smdawe looks good now! Glad you enjoyed your first dip into Kotlin 😄

**Pull request merged**